x-pack/plugins/security_solution/public/flyout/shared/components/flyout_navigation.tsx

```diff
@@ -79,7 +79,7 @@ export const FlyoutNavigation: FC<FlyoutNavigationProps> = memo(
       () => (
         <EuiButtonEmpty
           iconSide="left"
-          onClick={expandDetails}
+          onClick={expandDetails ? () => expandDetails() : undefined}
```

Comment:
just curious about why making this change here?
Also do you think you could add a small unit test for this change?

Reply:
Sure, I will add the test. This change is because the TS type for this function says that it doesn't have any parameters, but the `onClick` call has parameters. That introduced a bug in my component, and I decided to address the root cause. Do you think that we should update the type instead?

Reply:
thanks for adding a test and thanks for the explanation! I think what you have now is fine :)

Reply:
I was able to reproduce the bug locally, I'm getting a maximum call stack exceeded error, is that what you were getting as well?
![Screenshot 2023-12-13 at 4 34 34 PM](https://github.com/elastic/kibana/assets/17276605/ea6c085f-4837-46c9-a2be-9ad7fc2126cc)

I investigated a bit as I could not understand why your code would throw this error while [this one](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/public/flyout/document_details/right/navigation.tsx#L41) would not cause any issues (for example).
It seems that the problem comes from the `tab` variable you're passing [here](https://github.com/machadoum/kibana/blob/siem-ea-8038-tech-debt/x-pack/plugins/security_solution/public/flyout/entity_details/user_right/index.tsx#L94). If I console log the value, I get
![Screenshot 2023-12-13 at 4 45 18 PM](https://github.com/elastic/kibana/assets/17276605/0cc2014a-d374-4b49-8367-52391fb89710)
while it should only be a `string`. The problem then happens when we try to get the value in the url (within the `rison` library). You need to find a way to only pass the minimum amount of information, and looking at [this line](https://github.com/machadoum/kibana/blob/siem-ea-8038-tech-debt/x-pack/plugins/security_solution/public/flyout/entity_details/user_detais_left/index.tsx#L64) you should only need a string...

Also while I was testing, I tried to have the `tab` value as `undefined` as you have the fallback [here](https://github.com/machadoum/kibana/blob/siem-ea-8038-tech-debt/x-pack/plugins/security_solution/public/flyout/entity_details/user_right/index.tsx#L94) and the frontend crashes, so you might want to test that and makes sure your code handles this case correctly!

Reply:
I can quickly fix this by passing the `openPanel` prop as `expandDetails={() => openPanel()}`. But we still have the problem of the `expandDetails` type being wrong. 

```ts
export interface FlyoutNavigationProps {
  ...
  expandDetails?: () => void;
  ...
}
```
According to the type, it should not receive props, but it does.
Should I also update the `expandDetails ` prop?